cfme/scripting/bz.py

```diff
@@ -0,0 +1,186 @@
+    returns all BZs
+    """
+    BZTestCoverage = namedtuple("BZTestCoverage", ["id", "qe_test_coverage"])
+
+    bz_list = []
+    for bug_id in info.keys():
+        if open_only and info[bug_id]["is_open"] == "false":
+            continue
+        # get qe_test_coverage_flag
+        qe_test_coverage = "?"
+        for flag in info[bug_id]["flags"]:
+            if flag["name"] == "qe_test_coverage":
+                qe_test_coverage = flag["status"]
+                break
+        # append the BZ
+        bz_list.append(BZTestCoverage(id=bug_id, qe_test_coverage=qe_test_coverage))
+
+    return bz_list
+
+
+def cleanup():
+    click.echo("Removing the BZ report file, bz-report.yaml")
+    try:
+        os.remove("bz-report.yaml")
+    except OSError:
+        logger.exception("bz-report.yaml not found")
+
+
+@click.group(help="Functions for generating reports on BZs included in test suite metadata")
+def main():
+    pass
+
+
+@main.command(help="Generate BZ report on BZs that have coverage given a directory")
+@click.argument("directory", default="cfme/tests/")
+def report(directory):
+    get_report(directory)
+
+
+@main.command(help="List open/closed BZs that have test coverage")
+@click.argument("directory", default="cfme/tests/")
+@click.option(
+    "--all",
+    "-a",
+    "bz_status",
+    is_flag=True,
+    help="list all BZs with coverage",
+    default=True,
+    show_default=True,
+    flag_value="all_bzs",
+)
+@click.option(
+    "--open",
+    "-o",
+    "bz_status",
+    is_flag=True,
+    help="list open BZs with coverage",
+    default=False,
+    show_default=True,
+    flag_value="open_bzs"
+)
+@click.option(
+    "--closed",
+    "-c",
+    "bz_status",
+    is_flag=True,
+    help="list closed BZs with coverage",
+    default=False,
+    show_default=True,
+    flag_value="closed_bzs"
+)
+def list(directory, bz_status):
+    info = get_report(directory)
+
+    # get dict of information
+    status = STATUS[bz_status]
+
+    if status["val"]:
+        bz_list = [bug_id for bug_id in info.keys() if info[bug_id]["is_open"] == status["val"]]
+    else:
+        bz_list = [bug_id for bug_id in info.keys()]
+
+    if bz_list:
+        click.echo("The following BZ's {}: \n{}".format(status["text"], "\n".join(bz_list)))
+    else:
+        click.echo("I found no BZ's that {} BZ's".format(status["text"]))
+    cleanup()
+
+
+@main.command(help="Set QE test coverage flag based on automates/coverage metadata")
+@click.argument("directory", default="cfme/tests")
+@click.option(
+    "-s",
+    "--set",
+    "set_bzs",
+    is_flag=True,
+    help="Set QE test coverage on BZs that are marked in coverage/auotmates test metadata",
+    default=False,
+    show_default=True
+)
+def coverage(directory, set_bzs):
+    info = get_report(directory)
+
+    # get list of open BZs that are open and should have test coverage set
+    bz_list = get_qe_test_coverage(info, open_only=True)
```

Comment:
**Required** I think that we shouldn't just restrict this to open bugs. What if the user wants to set the coverage for a bug that has been closed.

Reply:
I'm going to implement something similar to the `miq list bz` so that the user can choose to audit `all`, only `open`, or only `closed` bzs.